include/deal.II/grid/grid_generator.h

```diff
@@ -316,6 +354,30 @@ namespace GridGenerator
                              const Point<dim>   (&corners) [dim],
                              const bool           colorize = false);
 
+  /**
+   * A subdivided parallelepiped.
+   *
+   * @param tria The Triangulation to create. It needs to be empty upon calling this
+   * function.
+   *
+   * @param origin First corner of the parallelepiped.
+   *
+   * @param axes A vector of @p dim tensors describing the length and direction of the edges
+   * from @p origin.
+   *
+   * @param subdivisions Number of subdivisions in each of the dim directions. Each
+   * entry must be positive. An empty vector is equivalent to one subdivision in
+   * each direction.
+   *
+   * @param colorize Assign different boundary ids if set to true.
+   */
+  template <int dim, int spacedim>
+  void
+  subdivided_parallelepiped (Triangulation<dim, spacedim>  &tria,
+                             const Point<spacedim> &origin,
+                             const std::vector<Tensor<1,spacedim> > &axes,
```

Comment:
Since you know the number of elements, you could make this argument of type
```
  Point<spacedim> (&axes)[dim]
```
or, if you like this better,
  const std_cxx11::array<Point<spacedim>,dim> &axes,
```

I don't particular like the name "axes" for this argument. In my mind, the axes of a parallelepiped connect the midpoints of the faces. But you think of these as originating at the "origin". Maybe call them `edge_vectors`? What do you think?